### ISSUE TITLE
appimage doesn't render preview area in layout/sequencer tabs

### ISSUE THREAD

**Issue by ohthehugemanatee** [smeighan/xLights#3117]
**Describe the bug**
Running the latest AppImage, the layout and sequence tabs have blank areas where the preview canvas should be. I cannot interact with that area at all.

**To Reproduce**
1. Download and chmod+x AppImage
2. Start xlights, create a new empty Show directory
3. Navigate to the layout tab. 

**Expected behavior**
Layout preview should be visible and allow interaction. Same goes for the sequence.

**Screenshots**
![image](https://user-images.githubusercontent.com/907257/152228608-077981af-d5df-4856-a0e3-0a91a649dd1e.png)

**Versions (please complete the following information):**
 - OS: Manjaro Linux 5.10.93-1
 - xLights version 2022.03 64bit
 - wxWidgets 3.1.6
 - sway 1.6.1 with xwayland
 
I found a forum post from someone complaining of the same symptom when they ran using their integrated Intel GPU. So in case it's relevant, I'm on a [Dell XPS 13 7390](https://wiki.archlinux.org/title/Dell_XPS_13_2-in-1_\(7390\)) with a 10th generation Intel Ice Lake processor with Iris Plus integrated graphics.

**Additional context**
I've tried:

- the previous AppImage version
- forcing OpenGL version 1.x/2.x
- 3d on/off
- setting a background image
- unsetting GTK-related environment vars `XDG_SESSION_TYPE` and  `XDG_CURRENT_DESKTOP` (normally set to `wayland` and `sway`, respectively).
- running on a fresh reboot

I tried but failed to compile myself on wxwidgets errors. And the docker compiled version wouldn't run because `error while loading shared libraries: libavformat.so.57: cannot open shared object file: No such file or directory`.

**Attachments**
debug console output:

```
❯ xlights --debug  
./src/common/stdpbase.cpp(59): assert ""traits"" failed in Get(): create wxApp before calling this
./src/common/stdpbase.cpp(59): assert ""traits"" failed in Get(): create wxApp before calling this
Unknown long option 'debug'
Usage: AppRun [-h] [--verbose]
  -h, --help	show this help message
  --verbose 	generate verbose log messages
Gtk-Message: 21:27:55.310: GtkDialog mapped without a transient parent. This is discouraged.

(AppRun.wrapped:24910): Gtk-CRITICAL **: 21:27:58.487: gtk_gesture_zoom_new: assertion 'GTK_IS_WIDGET (widget)' failed

(AppRun.wrapped:24910): Gtk-CRITICAL **: 21:27:58.487: gtk_event_controller_set_propagation_phase: assertion 'GTK_IS_EVENT_CONTROLLER (controller)' failed

(AppRun.wrapped:24910): GLib-GObject-WARNING **: 21:27:58.487: invalid (NULL) pointer instance

(AppRun.wrapped:24910): GLib-GObject-CRITICAL **: 21:27:58.487: g_signal_connect_data: assertion 'G_TYPE_CHECK_INSTANCE (instance)' failed

(AppRun.wrapped:24910): GLib-GObject-WARNING **: 21:27:58.487: invalid (NULL) pointer instance

(AppRun.wrapped:24910): GLib-GObject-CRITICAL **: 21:27:58.487: g_signal_connect_data: assertion 'G_TYPE_CHECK_INSTANCE (instance)' failed

(AppRun.wrapped:24910): GLib-GObject-WARNING **: 21:27:58.487: invalid (NULL) pointer instance

(AppRun.wrapped:24910): GLib-GObject-CRITICAL **: 21:27:58.487: g_signal_connect_data: assertion 'G_TYPE_CHECK_INSTANCE (instance)' failed

(AppRun.wrapped:24910): GLib-GObject-WARNING **: 21:27:58.487: invalid (NULL) pointer instance

(AppRun.wrapped:24910): GLib-GObject-CRITICAL **: 21:27:58.487: g_signal_connect_data: assertion 'G_TYPE_CHECK_INSTANCE (instance)' failed

(AppRun.wrapped:24910): Gtk-CRITICAL **: 21:27:58.487: gtk_widget_add_events: assertion 'GTK_IS_WIDGET (widget)' failed

(AppRun.wrapped:24910): GLib-GObject-WARNING **: 21:27:58.487: invalid (NULL) pointer instance

(AppRun.wrapped:24910): GLib-GObject-CRITICAL **: 21:27:58.487: g_signal_connect_data: assertion 'G_TYPE_CHECK_INSTANCE (instance)' failed

(AppRun.wrapped:24910): Gtk-CRITICAL **: 21:27:58.496: gtk_gesture_zoom_new: assertion 'GTK_IS_WIDGET (widget)' failed

(AppRun.wrapped:24910): Gtk-CRITICAL **: 21:27:58.496: gtk_event_controller_set_propagation_phase: assertion 'GTK_IS_EVENT_CONTROLLER (controller)' failed

(AppRun.wrapped:24910): GLib-GObject-WARNING **: 21:27:58.496: invalid (NULL) pointer instance

(AppRun.wrapped:24910): GLib-GObject-CRITICAL **: 21:27:58.496: g_signal_connect_data: assertion 'G_TYPE_CHECK_INSTANCE (instance)' failed

(AppRun.wrapped:24910): GLib-GObject-WARNING **: 21:27:58.496: invalid (NULL) pointer instance

(AppRun.wrapped:24910): GLib-GObject-CRITICAL **: 21:27:58.496: g_signal_connect_data: assertion 'G_TYPE_CHECK_INSTANCE (instance)' failed

(AppRun.wrapped:24910): GLib-GObject-WARNING **: 21:27:58.496: invalid (NULL) pointer instance

(AppRun.wrapped:24910): GLib-GObject-CRITICAL **: 21:27:58.496: g_signal_connect_data: assertion 'G_TYPE_CHECK_INSTANCE (instance)' failed

(AppRun.wrapped:24910): GLib-GObject-WARNING **: 21:27:58.496: invalid (NULL) pointer instance

(AppRun.wrapped:24910): GLib-GObject-CRITICAL **: 21:27:58.496: g_signal_connect_data: assertion 'G_TYPE_CHECK_INSTANCE (instance)' failed

(AppRun.wrapped:24910): Gtk-CRITICAL **: 21:27:58.496: gtk_widget_add_events: assertion 'GTK_IS_WIDGET (widget)' failed

(AppRun.wrapped:24910): GLib-GObject-WARNING **: 21:27:58.496: invalid (NULL) pointer instance

(AppRun.wrapped:24910): GLib-GObject-CRITICAL **: 21:27:58.496: g_signal_connect_data: assertion 'G_TYPE_CHECK_INSTANCE (instance)' failed

(AppRun.wrapped:24910): Gtk-CRITICAL **: 21:27:59.257: gtk_gesture_zoom_new: assertion 'GTK_IS_WIDGET (widget)' failed

(AppRun.wrapped:24910): Gtk-CRITICAL **: 21:27:59.257: gtk_event_controller_set_propagation_phase: assertion 'GTK_IS_EVENT_CONTROLLER (controller)' failed

(AppRun.wrapped:24910): GLib-GObject-WARNING **: 21:27:59.257: invalid (NULL) pointer instance

(AppRun.wrapped:24910): GLib-GObject-CRITICAL **: 21:27:59.257: g_signal_connect_data: assertion 'G_TYPE_CHECK_INSTANCE (instance)' failed

(AppRun.wrapped:24910): GLib-GObject-WARNING **: 21:27:59.257: invalid (NULL) pointer instance

(AppRun.wrapped:24910): GLib-GObject-CRITICAL **: 21:27:59.257: g_signal_connect_data: assertion 'G_TYPE_CHECK_INSTANCE (instance)' failed

(AppRun.wrapped:24910): GLib-GObject-WARNING **: 21:27:59.257: invalid (NULL) pointer instance

(AppRun.wrapped:24910): GLib-GObject-CRITICAL **: 21:27:59.257: g_signal_connect_data: assertion 'G_TYPE_CHECK_INSTANCE (instance)' failed

(AppRun.wrapped:24910): GLib-GObject-WARNING **: 21:27:59.257: invalid (NULL) pointer instance

(AppRun.wrapped:24910): GLib-GObject-CRITICAL **: 21:27:59.257: g_signal_connect_data: assertion 'G_TYPE_CHECK_INSTANCE (instance)' failed

(AppRun.wrapped:24910): Gtk-CRITICAL **: 21:27:59.257: gtk_gesture_rotate_new: assertion 'GTK_IS_WIDGET (widget)' failed

(AppRun.wrapped:24910): Gtk-CRITICAL **: 21:27:59.257: gtk_event_controller_set_propagation_phase: assertion 'GTK_IS_EVENT_CONTROLLER (controller)' failed

(AppRun.wrapped:24910): GLib-GObject-WARNING **: 21:27:59.257: invalid (NULL) pointer instance

(AppRun.wrapped:24910): GLib-GObject-CRITICAL **: 21:27:59.257: g_signal_connect_data: assertion 'G_TYPE_CHECK_INSTANCE (instance)' failed

(AppRun.wrapped:24910): GLib-GObject-WARNING **: 21:27:59.257: invalid (NULL) pointer instance

(AppRun.wrapped:24910): GLib-GObject-CRITICAL **: 21:27:59.257: g_signal_connect_data: assertion 'G_TYPE_CHECK_INSTANCE (instance)' failed

(AppRun.wrapped:24910): GLib-GObject-WARNING **: 21:27:59.257: invalid (NULL) pointer instance

(AppRun.wrapped:24910): GLib-GObject-CRITICAL **: 21:27:59.257: g_signal_connect_data: assertion 'G_TYPE_CHECK_INSTANCE (instance)' failed

(AppRun.wrapped:24910): GLib-GObject-WARNING **: 21:27:59.257: invalid (NULL) pointer instance

(AppRun.wrapped:24910): GLib-GObject-CRITICAL **: 21:27:59.257: g_signal_connect_data: assertion 'G_TYPE_CHECK_INSTANCE (instance)' failed

(AppRun.wrapped:24910): Gtk-CRITICAL **: 21:27:59.257: gtk_widget_add_events: assertion 'GTK_IS_WIDGET (widget)' failed

(AppRun.wrapped:24910): GLib-GObject-WARNING **: 21:27:59.257: invalid (NULL) pointer instance

(AppRun.wrapped:24910): GLib-GObject-CRITICAL **: 21:27:59.257: g_signal_connect_data: assertion 'G_TYPE_CHECK_INSTANCE (instance)' failed

(AppRun.wrapped:24910): Gtk-WARNING **: 21:27:59.558: Negative content width -11 (allocation 1, extents 6x6) while allocating gadget (node button, owner GtkButton)

(AppRun.wrapped:24910): Gtk-CRITICAL **: 21:27:59.558: gtk_box_gadget_distribute: assertion 'size >= 0' failed in GtkScrollbar

(AppRun.wrapped:24910): Gtk-CRITICAL **: 21:27:59.558: gtk_widget_get_preferred_width_for_height: assertion 'height >= 0' failed

(AppRun.wrapped:24910): Gtk-WARNING **: 21:27:59.558: gtk_widget_size_allocate(): attempt to allocate widget with width 13 and height -7

(AppRun.wrapped:24910): Gtk-CRITICAL **: 21:27:59.558: gtk_box_gadget_distribute: assertion 'size >= 0' failed in GtkScrollbar

(AppRun.wrapped:24910): Gtk-WARNING **: 21:27:59.579: Negative content width -9 (allocation 25, extents 17x17) while allocating gadget (node button, owner GtkButton)

(AppRun.wrapped:24910): Gtk-CRITICAL **: 21:27:59.593: gtk_box_gadget_distribute: assertion 'size >= 0' failed in GtkScrollbar

(AppRun.wrapped:24910): Gtk-CRITICAL **: 21:27:59.594: gtk_box_gadget_distribute: assertion 'size >= 0' failed in GtkScrollbar

(AppRun.wrapped:24910): Gtk-CRITICAL **: 21:27:59.595: gtk_box_gadget_distribute: assertion 'size >= 0' failed in GtkCheckButton

(AppRun.wrapped:24910): Gtk-CRITICAL **: 21:27:59.595: gtk_box_gadget_distribute: assertion 'size >= 0' failed in GtkScrollbar

(AppRun.wrapped:24910): Gtk-CRITICAL **: 21:27:59.595: gtk_box_gadget_distribute: assertion 'size >= 0' failed in GtkScrollbar

(AppRun.wrapped:24910): Gtk-CRITICAL **: 21:27:59.595: gtk_box_gadget_distribute: assertion 'size >= 0' failed in GtkScrollbar

(AppRun.wrapped:24910): Gtk-CRITICAL **: 21:27:59.596: gtk_box_gadget_distribute: assertion 'size >= 0' failed in GtkScrollbar
```

Thank you for your work on this awesome application, and for your help!

**Comment by ManiacTwister** [comment]
I can confirm this issue. It worked until i updated my system to latest manjaro. (Haven't updated since several months)

**Comment by ohthehugemanatee** [comment]
I had to reinstall on Ubuntu 22.04 for work. Set up swaywm with the same configs, and xlights works fine. So there's likely something in the version diff. On Ubuntu I'm running:

sway:
```
Version: 1.6.1-2
Depends: libgl1-mesa-dri, swaybg, policykit-1, libc6 (>= 2.34), libcairo2 (>= 1.10.0), libevdev2 (>= 0.9.1), libgdk-pixbuf-2.0-0 (>= 2.31.1), libgles2, libglib2.0-0 (>= 2.12.0), libinput10 (>= 1.6.0), libjson-c5 (>= 0.15), libpango-1.0-0 (>= 1.22.0), libpangocairo-1.0-0 (>= 1.22.0), libpcre3, libpixman-1-0 (>= 0.25.2), libsystemd0 (>= 243), libudev1 (>= 183), libwayland-client0 (>= 1.9.91), libwayland-cursor0 (>= 1.0.2), libwayland-server0 (>= 1.14.91), libwlroots9 (>= 0.14.0), libxcb1, libxkbcommon0 (>= 0.7.0)
```
libwlroots:
```
Version: 0.14.1-5
Depends: libc6 (>= 2.34), libdrm2 (>= 2.4.105), libegl1, libgbm1 (>= 21.1.0), libgles2, libinput10 (>= 1.9.0), libpixman-1-0 (>= 0.27.4), libseat1 (>= 0.5.0), libudev1 (>= 183), libwayland-client0 (>= 1.19), libwayland-server0 (>= 1.19), libxcb-composite0, libxcb-dri3-0 (>= 1.13), libxcb-icccm4 (>= 0.4.1), libxcb-present0, libxcb-render-util0, libxcb-render0, libxcb-res0 (>= 1.10), libxcb-shm0 (>= 1.10), libxcb-xfixes0, libxcb-xinput0 (>= 1.14), libxcb1, libxkbcommon0 (>= 0.5.0)
```

Any other version info that might be relevant here?

**Comment by reenx** [comment]
I am also hit by this problem after updating my system. I am using Fedora 35. Any progress or ideas to investigate?

**Comment by reenx** [comment]
I am able to compile xLights on my system now. That version works without problems. To get it to work I had to modify the makefile to use lua in stead of lua53. And in the sol.hpp file there was a reference to luaopen_bit32 that could not be resolved. I removed that and now it compiles and xLights seems to work.

So could the lua library somehow be the reason for the missing layout view?
Is there some way to analyze the AppImage file. Perhaps a library is missing from the AppImage file?

**Comment by reenx** [comment]
I updated my system to the latest fedora release (36). And with the 2022.09 appimage this problem is resolved for me.
Older versions still show the same problem. So I think the update of my system has nothing to do with this. 
But to summarize. For me this issue is resolved with 2022.09.
